omero/util_scripts/Move_Annotations.py

```diff
@@ -0,0 +1,234 @@
+    except Exception, ex:
+        log("Failed to create links: %s" % ex.message)
+        return 0
+
+    if remove_anns:
+        log("Deleting ImageAnnotation links... %s" % link_ids)
+        try:
+            for link_id in link_ids:
+                to_delete = ImageAnnotationLinkI(link_id, False)
+                conn.getUpdateService().deleteObject(to_delete)
+        except Exception, ex:
+            log("Failed to delete links: %s" % ex.message)
+    return len(new_links)
+
+
+def move_annotations(conn, script_params):
+    """Process script parameters and move annotations as specified."""
+    plates = []
+    filter_type = None
+
+    dtype = script_params['Data_Type']
+    ids = script_params['IDs']
+    ann_type = script_params['Annotation_Type']
+    if ann_type in ANN_TYPES:
+        filter_type = ANN_TYPES[ann_type]
+    if ann_type == 'Rating':
+        ns = NSINSIGHTRATING
+    else:
+        ns = script_params.get('Namespace')
+    remove_anns = script_params['Remove_Annotations_From_Images']
+
+    # Get the Plates or Wells
+    objects = list(conn.getObjects(dtype, ids))
+
+    ann_total = 0
+
+    if dtype == 'Well':
+        for well in objects:
+            ann_count = move_well_annotations(conn, well, filter_type,
+                                              remove_anns, ns)
+            ann_total += ann_count
+    else:
+        if dtype == 'Plate':
+            plates = objects
+        elif dtype == 'Screen':
+            for screen in objects:
+                plates.extend(list(screen.listChildren()))
+        log("Found Plates: %s" % [p.id for p in plates])
+        for plate in plates:
+            for well in plate.listChildren():
+                ann_count = move_well_annotations(conn, well, filter_type,
+                                                  remove_anns, ns)
+                ann_total += ann_count
+
+    return objects, ann_total
+
+
+def run_script():
+    """The main entry point of the script."""
+    data_types = [rstring('Screen'), rstring('Plate'), rstring('Well')]
+
+    ann_types = [rstring('All')]
+    ann_types.extend([rstring(k) for k in ANN_TYPES.keys()])
+
+    client = scripts.client(
+        'Move_Annotations.py',
+        """
+For Screen/Plate/Well data, this script moves your Annotations from Images to
+their parent Wells. If you are an Admin, this will also move annotations that
+other users have added, creating links that belong to the same users.
+    """,
+
+        scripts.String(
+            "Data_Type", optional=False, grouping="1",
+            description="The data you want to work with.", values=data_types,
+            default="Plate"),
+
+        scripts.List(
+            "IDs", optional=False, grouping="2",
+            description="List of Screen, Plate or Well IDs").ofType(rlong(0)),
+
+        scripts.String(
+            "Annotation_Type", grouping="3",
+            description="Move All annotations OR just one type of annotation",
```

Comment:
Ooops - getting a touch of Title Case here. https://medium.com/@jsaito/making-a-case-for-letter-case-19d09f653c98#.h1s476nc6. I'll fix...